src/main/java/org/candlepin/insights/controller/InventoryController.java

```diff
@@ -168,6 +164,28 @@ private void extractNetworkFacts(Map<String, String> pinheadFacts, ConduitFacts
         if (!macAddresses.isEmpty()) {
             facts.setMacAddresses(macAddresses);
         }
+        extractIpAddresses(pinheadFacts, facts);
+    }
+
+    private void extractIpAddresses(Map<String, String> pinheadFacts, ConduitFacts facts) {
+        Set<String> ipAddresses = new HashSet<>();
+        pinheadFacts.entrySet().stream()
+            .filter(entry -> entry.getKey().matches(IP_ADDRESS_FACT_REGEX) && !isEmpty(entry.getValue()))
+            .forEach(entry -> {
+                // The facts ending with _list have precedence.
+                // We might end up adding the same addresses twice,
+                // but Set<String> will take care of the duplicates.
+                String name = entry.getKey();
+                if (!name.endsWith(LIST_SUFFIX) && pinheadFacts.containsKey(name + LIST_SUFFIX)) {
+                    name = name + LIST_SUFFIX;
```

Comment:
Would you mind explaining a little what is going on here?  It looks like if we have a fact like "...global" or "...link" they don't ever get added at all since `name` gets mutated to add "_list" onto the end before we fetch the value from the map.  Maybe that's the desired behavior, but I just want to double-check.

Reply:
@sskracic after chatting w/ Al a bit, we realized that the non `_list` facts are always going to be redundant when present, so the set logic will take care of things, and we don't have to carry this extra LIST_SUFFIX logic. So let's just let the set handle all dedup.

Reply:
Ack - removed all the extra LIST_SUFFIX logic.